apaq-v1.0.yaml

```diff
@@ -0,0 +1,113 @@
+      description: |
+        Uploads airport information and predicts air quality based on the provided data.
+        Initiates background processing for prediction.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/definitions/AirportData'
+      responses:
+        '201':
+          description: |
+            An air quality prediction has successfully been created and returned.
+          schemas:
+            $ref: '#/defintions/PredictionResponse'
+          examples:
+            applications/json:
+              air_quality:
+                - 2.12
+        '400':
+          $ref: '#/responses/BadRequest'
+
+definitions:
+  AirportData:
+    type: object
+    properties:
+      altitude:
+        type: number
+        description: Altitude of the airport [ft]
+      runways:
+        type: array
+        description: A list of available runways at the specified airport
+        items:
+          type: object
+          properties:
+            length:
+              type: number
+              description: Length of the runway [ft]
+            surface:
+              type: string
+              description: Surface material of the runway
+      total_arrivals:
+        type: number
+        description: Total number of arriving flights in the next 24 hours.
+      total_departures:
+        type: number
+        description: Total number of departing flights in the next 24 hours.
+
+  PredictionResponse:
+    type: object
+    properties:
+      air_quality:
+        type: number
+        description: Predicted air quality value [PM10]
+
+  Error:
+    type: object
+    description: |
+      General error response body
+    properties:
+      errors:
+        description: |
+          This field MUST contain an array with at least one error model.
+        type: array
+        items:
+          $ref: '#/definitions/ErrorItem'
+        minItems: 1
+      status_code:
+        description: |
+          This field MAY contain the HTTP status code used for the response. Otherwise, it MUST be omitted.
+        type: integer
+    required:
+      - errors
+    additionalProperties: false
+
+  ErrorItem:
+    description: |
+      An error message.
+    type: object
+    properties:
+      code:
+        description: |
+          This field MUST contain a snake case string succinctly identifying the problem.
+          This field SHOULD NOT indicate which field, parameter, or header caused the error.
+        type: string
+      message:
+        description: |
+          This field MUST contain a plainly-written, developer-oriented explanation of the solution to the problem in complete, well-formed sentences.
+        type: string
+    required:
+      - code
+      - message
+    additionalProperties: false
+
+responses:
+  BadRequest:
+    description: |
+      The request body or parameters were incorrect
+  schema:
```

Comment:
These should be in the `errors` section of `components` and perhaps it would be better to label them with the code as well?